packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -564,37 +578,93 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         const points: any = [];
         let zeros = 0;
 
         this.state.xyData.datasets.forEach((d: any) => {
-            // In case there are less data points than pixels in the chart,
-            // calculate nearest value.
-            const yValue = d.data[index];
-            const rounded = isNaN(yValue) ? 0 : (Math.round(Number(yValue) * 100) / 100);
-            // If there are less than 10 lines in the chart, show all values, even if they are equal to 0.
-            // If there are more than 10 lines in the chart, summarise the ones that are equal to 0.
-            if (this.state.xyData.datasets.length <= 10 || rounded > 0) {
-                const point: any = {
-                    label: d.label,
-                    color: d.borderColor,
-                    background: d.backgroundColor,
-                    value: rounded.toString(),
-                };
-                points.push(point);
+            let yValue = 0;
+            let xValue = 0;
+            let invalidPoint = false;
+            if (this.isScatterPlot && d.data.length > 0) {
+                // Find the data point that is the closest to the mouse position
+                const point = { x: Number(timeForX), y: y - chartTop };
+                const closest = this.closestPoint(d.data, point, xRatio, yRatio, chartHeight);
+                if (closest !== undefined) {
+                    yValue = d.data[closest].y;
+                    xValue = d.data[closest].x;
+                }
+                else {
+                    invalidPoint = true; // Too far from mouse
+                }
             }
             else {
-                zeros += 1;
+                // In case there are less data points than pixels in the chart,
+                // calculate nearest value.
+                yValue = d.data[index];
+            }
+            const rounded: number = isNaN(yValue) ? 0 : (Math.round(Number(yValue) * 100) / 100);
+            let formatted: string = new Intl.NumberFormat().format(rounded);
+            if (this.isScatterPlot && this.props.unitController.numberTranslator) {
+                const time = this.props.unitController.numberTranslator(BigInt(xValue));
+                formatted = '(' + time + ') ' + formatted;
+                timeLabel = 'Process (time stamp) value';
+            }
+
+            // If there are less than 10 lines in the chart, show all values, even if they are equal to 0.
+            // If there are more than 10 lines in the chart, summarise the ones that are equal to 0.
+            if (!invalidPoint) {
+                if (this.state.xyData.datasets.length <= 10 || rounded > 0) {
+                    const point: any = {
+                        label: d.label,
+                        color: d.borderColor,
+                        background: d.backgroundColor,
+                        value: formatted
+                    };
+                    points.push(point);
```

Comment:
Can you only create the point and push it, if it's closer than the previous point?

Reply:
![Screenshot from 2022-01-27 15-51-30](https://user-images.githubusercontent.com/419535/151444920-036f2eee-6911-40f5-b277-c8d69773f134.png)
Look at all that GC! :)

Reply:
Yes, thank you for the help there.